src/app/swap/page.tsx

```diff
@@ -74,12 +79,12 @@ export default function Swap() {
   const { setModalType, payload, onCloseModal } = useModalStore(
     (state) => state
   )
-  const { bestEstimate, allEstimates, getSwapEstimateBot, isFetching } =
+  const { bestEstimate, allEstimates, getSwapEstimateBot } =
     useSwapEstimateBot()
   const isProgrammaticUpdate = useRef(false)
   useModalSearchParams()
-  const { handleValidateInputs, errorMsg } = useSwapGuard()
-
+  const [errorMsg, setErrorMsg] = useState<ErrorEnum>()
+  const [isFetching, setIsFetching] = useState(false)
```

Comment:
Fetching state is purely UI conect so lets keep on UI form level

Reply:
We have already use `react-hook-form` with error handlers which works behind the scene. And it gives you all needed instruments like form schemas and different validation rules without needs to write all this thins manually.

Reply:
This can't be caught with react-hook-form, that is why it should be handled like this.

P.S. I think `react-hook-form` is rebundant for such a simple form

Reply:
We use `useSwapGuard` for all such cases. Please check relevant comment below.

Reply:
We can also move all the form logic to a dedicated component, for example `SwapForm.tsx`. You can use this link as an example - https://github.com/defuse-protocol/defuse-sdk/blob/feat/swap/src/features/swap/components/SwapForm.tsx.

In this case, you need to create a folder `components` by this path `src/app/swap/`

Reply:
In order don't stop the progress, let's save all this for now and move the form logic into a separate component in the next release.